### PR TITLE
[new release] irmin (19 packages) (3.10.0)

### PR DESCRIPTION
**Pull request by samoht**
Irmin, a distributed database that follows the same design principles as Git

- Project page: <a href="https://github.com/mirage/irmin">https://github.com/mirage/irmin</a>
- Documentation: <a href="https://mirage.github.io/irmin/">https://mirage.github.io/irmin/</a>

### Added

- **irmin-git**
  - Expose `Content_addressable` type (mirage/irmin#2329, @art-w)

### Changed

- **irmin**
  - Rename `Node.S.effect` to `read_effect` for OCaml 5.3 compatibility (mirage/irmin#2347, @art-w)

### Fixed

- **irmin-client**
  - Fix a fd leak when using `clone` (mirage/irmin#2322, @samoht)
- **irmin-git**
  - Fix git sync example (mirage/irmin#2327, @art-w)
  - Fixed issue with two subsequent `Irmin-git` repos, where one
    repo would lose the ability to find its root path (mirage/irmin#2326, @zazedd)
- **irmin**
  - Fix CI, update dependencies (mirage/irmin#2321, @smorimoto)
  - Update documentation (mirage/irmin#2323, mirage/irmin#2324, mirage/irmin#2325, @christinerose)
- **irmin-cli**
  - Fix uncaught exception (mirage/irmin#2326, @art-w)

### Removed

- **irmin**
  - Removed `?allow_duplicate` parameter from the `Conf.key` function (mirage/irmin#2326, @zazedd)

### CODE REVIEW

**Comment by samoht** [comment]
I'm fixing the lint errors here: https://github.com/mirage/irmin/pull/2350

File: packages/irmin-client/irmin-client.3.10.0/opam [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  "fmt" {>= "0.9.0"}
+  "logs" {>= "0.7.0"}
+  "lwt" {>= "5.7.0"}
+  "irmin-test" {= version & with-test}
+  "alcotest-lwt" {with-test & >= "1.8.0"}
+  "irmin-watcher" {with-test & >= "0.5.0"}
+  "mirage-crypto-rng-lwt" {with-test & >= "1.1.0"}
+]
+build: [
+  ["dune" "subst"] {pinned}
```
Comment (samoht) [inline packages/irmin-client/irmin-client.3.10.0/opam:30]:
```suggestion
  ["dune" "subst"] {dev}
```

File: packages/irmin-server/irmin-server.3.10.0/opam [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  "cohttp-lwt-unix"
+  "ppx_blob" {>= "0.7.2"}
+  "digestif" {>= "1.1.4"}
+  "alcotest-lwt" {>= "1.7.0" & with-test}
+  "irmin-watcher" {>= "0.5.0" & with-test}
+  "irmin-test" {= version & with-test}
+]
+
+build: [
+  ["dune" "subst"] {pinned}
```
Comment (samoht) [inline packages/irmin-server/irmin-server.3.10.0/opam:33]:
```suggestion
  ["dune" "subst"] {dev}
```

**Comment by shonfeder** [comment]
The windows CI failures are from the (transitive) `rusage` dependency, so not relevant to anything introduced here.

**Comment by shonfeder** [comment]
One CI failure that may of interest on https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/2ecf2dcedf527dc5dd5dd6fd8c1583505cdab018/variant/compilers,5.3~beta1,irmin-client.3.10.0,tests

```
<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
Processing  2/3: [irmin-client: dune build]
+ /home/opam/.opam/opam-init/hooks/sandbox.sh "build" "dune" "build" "-p" "irmin-client" "-j" "39" (CWD=/home/opam/.opam/5.3~beta1/.opam-switch/build/irmin-client.3.10.0)
Processing  2/3: [irmin-client: dune runtest]
+ /home/opam/.opam/opam-init/hooks/sandbox.sh "build" "dune" "runtest" "-p" "irmin-client" (CWD=/home/opam/.opam/5.3~beta1/.opam-switch/build/irmin-client.3.10.0)
- File "test/irmin-client/dune", line 2, characters 7-11:
- 2 |  (name test)
-            ^^^^
- (cd _build/default/test/irmin-client && ./test.exe)
- Fatal error: exception Unix.Unix_error(Unix.ENAMETOOLONG, "", "/home/opam/.opam/5.3~beta1/.opam-switch/build/irmin-client.3.10.0/_build/default/test/irmin-client/test.sock")
- Fatal error: exception Unix.Unix_error(Unix.ENAMETOOLONG, "", "/home/opam/.opam/5.3~beta1/.opam-switch/build/irmin-client.3.10.0/_build/default/test/irmin-client/test.sock")
[ERROR] The compilation of irmin-client.3.10.0 failed at "dune runtest -p irmin-client".

#=== ERROR while compiling irmin-client.3.10.0 ================================#
# context              2.4.0~alpha1~dev | linux/x86_64 | ocaml-base-compiler.5.3.0~beta1 | pinned(https://github.com/mirage/irmin/releases/download/3.10.0/irmin-3.10.0.tbz)
# path                 ~/.opam/5.3~beta1/.opam-switch/build/irmin-client.3.10.0
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune runtest -p irmin-client
# exit-code            1
# env-file             ~/.opam/log/irmin-client-7-2547a7.env
# output-file          ~/.opam/log/irmin-client-7-2547a7.out
### output ###
# File "test/irmin-client/dune", line 2, characters 7-11:
# 2 |  (name test)
#            ^^^^
# (cd _build/default/test/irmin-client && ./test.exe)
# Fatal error: exception Unix.Unix_error(Unix.ENAMETOOLONG, "", "/home/opam/.opam/5.3~beta1/.opam-switch/build/irmin-client.3.10.0/_build/default/test/irmin-client/test.sock")
# Fatal error: exception Unix.Unix_error(Unix.ENAMETOOLONG, "", "/home/opam/.opam/5.3~beta1/.opam-switch/build/irmin-client.3.10.0/_build/default/test/irmin-client/test.sock")
```

**Comment by shonfeder** [comment]
> Is this expected or accounted for?

Ah, I see this is just a case of https://github.com/ocaml/opam-repository/pull/14252#issuecomment-499541925 and only has to do with limits on the length of filenemes for sockets on certain platforms.

So the CI looks good to me than. I think this is good to merge, if desired, future releases could try to generate shorter socket names for tests to avoid this noise.

Thanks for the updates!

**Pull request merged**